apollo-federation/src/supergraph/mod.rs

```diff
@@ -1474,6 +1533,35 @@ fn add_subgraph_field(
         )?;
     }
 
+    if let Some(context_arguments) = &field_directive_application.context_arguments {
+        for args in context_arguments {
+            let ContextArgument {
+                name,
+                type_,
+                context,
+                selection,
+            } = args;
+            let mut split = context.split("__");
+            split.next();
+            let Some(context_name_in_subgraph) = split.last() else {
+                todo!();
+            };
```

Comment:
This is my mistake. There is an nicer way of expressing this via [`str::rsplit_once`](https://doc.rust-lang.org/std/primitive.str.html#method.rsplit_once), which returns `None` if the string doesn't contain the substring you're looking for.
```suggestion
            let Some((_, context_name_in_subgraph)) = context.rsplit_once() else {
                todo!();
            };
```